fdbserver/VersionedBTree.actor.cpp

```diff
@@ -2139,7 +2139,7 @@ class DWALPager final : public IPager2 {
 	          int desiredExtentSize,
 	          std::string filename,
 	          int64_t pageCacheSizeBytes,
-	          Version remapCleanupWindow,
+	          int64_t remapCleanupWindow,
```

Comment:
The meaning of `remapCleanupWindow` is not consistent between usage in `DWALPager` and initializations.  
- It is being passed as a Byte count for most calls to the `DWALPager()` constructor except for...
- In `KeyValueStoreRedwood()` it is passed as an Entries count, calculated via the desired page size.
- Internally in `DWALPager` it is treated as an Entries count in the remap cleanup logic.

It actually needs to be Bytes everywhere and both the `DWALPager()` parameter here and the `remapCleanupWindow` member should renamed to `remapCleanupWindowBytes`.  Then in the `remapCleanup()` actor you can convert the Bytes to Entries for local variables using the _actual_ page size which is a member of `DWALPager`.

The reason for this is that the page size passed to `DWALPager` is only a "desired" page size as noted by the parameter name, because if the `filename` already exists on disk then its native page size must be used instead of the desired page size in the constructor parameter.  Therefore, it is not possible to know the current Entries count for the remap queue limit outside of `DWALPager`, at least not without inspecting the file first if it exists.  The most straightforward approach is to specify Bytes and let remap cleanup, which only called after recovery so the page size is known, handle Entries count calculation.

Reply:
fixed